pyproject.toml

```diff
@@ -8,6 +8,74 @@ requires = [
 # Defined by PEP 517
 build-backend = "setuptools.build_meta"
 
+[project]
+authors = [
+    {name = "Iris Contributors", email = "scitools.pub@gmail.com"}
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: GNU Lesser General Public License v3 or later (LGPLv3+)",
+    "Operating System :: MacOS",
+    "Operating System :: POSIX",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: Unix",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Atmospheric Science",
+    "Topic :: Scientific/Engineering :: Visualization",
+]
+dynamic = [
+    "dependencies",
+    "optional-dependencies",
+    "readme",
+    "version",
```

Comment:
https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

>When these fields are expected to be provided by setuptools a corresponding entry is required in the `tool.setuptools.dynamic` table
>...
>```
>...
>[tool.setuptools.dynamic]
>version = {attr = "my_package.VERSION"}
>```

Do we therefore need to include a `version` further down, too? If setuptools_scm is doing something magic here it isn't clear to me.

Reply:
Yeah, I've hummed and hawed about this one... it's not entirely clear.

I've intentionally followed the pattern on `xesmf`, see https://github.com/pangeo-data/xESMF/pull/248

My understanding is that the `setuptools-scm` plugin can do it under the hood without explicitly specifying it, and it certainly seem to work.

I'll retry with the explicit `dynamic.version` metadata and see how it rolls...

Reply:
I also get problems, although they appear to be because it's trying to run `iris.__init__` in an environment without all the dependencies. Anyway I can confirm that build works fine without my suggestion.